dart-impl/mpi/src/dart_communication.c

```diff
@@ -193,19 +215,45 @@ dart_ret_t dart_get(
                    nelem, (unsigned long)win, team_unit_id.id, offset, dest);
   }
 
-
-  DART_LOG_TRACE("dart_get:  MPI_Get");
-  if (MPI_Get(dest,
-              nelem,
-              mpi_dtype,
-              team_unit_id.id,
-              offset,
-              nelem,
-              mpi_dtype,
-              win)
-      != MPI_SUCCESS) {
-    DART_LOG_ERROR("dart_get ! MPI_Rget failed");
-    return DART_ERR_INVAL;
+  /*
+   * MPI uses offset type int, do not copy more than INT_MAX elements:
+   */
+  // chunk up the get
+  const size_t nchunks   = nelem / INT_MAX;
```

Comment:
Everything looks good. However, the hard coded INT_MAX as the maximum number of contiguous elements is used in at least two difference compile units. It may change in future (for whatever reason). Maybe we add an explicit macro like MAX_CONTIG_ELEMENTS and define it to INT_MAX.